api/nomics.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+import calendar
+import connexion
+
+from services.bitshares_websocket_client import client as bitshares_ws_client
+from services.bitshares_elasticsearch_client import client as bitshares_es_client
+from services.cache import cache
+import api.explorer
+import es_wrapper
+import config
+
+
+def info():
+    return {
+        "name": "Bitshares",
+        "description": "An exchange description of at least 1000 characters in plain text (no html)",
+        "location": "Anywhere",
+        "logo": "https://example.com/exchange-logo.png",
+        "website": "https://bitshares.org/",
+        "twitter": "example",
+        "capability": {
+            "markets": True,
+            "trades": True,
+            "tradesSocket": False,
+            "orders": False,
+            "ordersSocket": False,
+            "ordersSnapshot": True,
+            "candles": False
+        }
+    }
+
+
+def markets():
+    result = []
+    top_markets = bitshares_ws_client.request('database', 'get_top_markets', [100])
```

Comment:
Alternatively `bitshares_es_client.get_markets(from_date="now-1d", to_date="now")` could be used.
It retrieves all the markets that have at least one trade.
Then we need to merge ways (BTS/CNY + CNY/BTS) and we can use number of orders to sort (instead of base_volume as get_top_markets does wich is not comparable from one market to another)

Reply:
Ill try this.

Reply:
One problem i can see with this approach is that the ES call will respond with asset ids, we need to convert each of them to name plus doing some more processing(merge, sort, etc). Doing this with the `api.explorer.get_asset` will be too much even considering the call is cached. First time will take ages, next ones should take less. What do you think about that ?

Reply:
Yes it might be a slower query (a couple of seconds) compared to core.
Note that it's already used for `get_most_active_markets`/`get_top_markets` for the explorer. So the cache could be shared. You can also already have an idea of the time it takes. 
The computation itself could be negligible, what takes time is the ES query.
About the `get_asset` issue it's the same story.. it is already used all over the place, so we share the cache.
Nomics should not have to query this endpoint very often. To be checked.

Reply:
i honestly think it will not be a good idea to use this right now. on localhost a call to `get_most_active_markets` the first time takes like a minute and online it is timing out: http://148.251.10.231:5000/apidocs/#!/market/api_explorer_get_most_active_markets

why not just use api `get_top_markets` that takes no time for this ? i understand the results are not maybe the best but i will prefer to just hardcode the markets we want to show than giving them a query that takes that long, this is specific for nomics

Reply:
`get_top_markets` is fine for me.

May be we could improve it to sort by number of traders or convert all volumes to a comparable currency. Ideally, it should be a parameter of the query.

Reply:
> online it is timing out: http://148.251.10.231:5000/apidocs/#!/market/api_explorer_get_most_active_markets

We should open an issue to move it to `get_top_markets`... ;-)

Reply:
i agree that the node api call can be better but what they request IMO at: https://github.com/nomics-crypto/nomics-platform/blob/master/doc/cryptocurrency-api-exchange-integration.md#markets---available-markets---required is just a list of all markets, we dont know how they will sort the list, we want the important markets to be there.

this is fast and good enough for the requirement if you ignore the order: http://185.208.208.184:5000/nomics/markets

Reply:
Pull request in progress to extend `get_top_markets` in the core at https://github.com/bitshares/bitshares-core/pull/1911 
We will follow this path and probably do not change to ES to get this results but keep using `get_top_markets` improved.